### PR TITLE
py-cupy allow customizing architecture and threads

### PR DESCRIPTION
**Pull request by robertu94**
@adamjstewart 

This should in theory allow setting which architectures to build with, but doesn't seem to work.  As far as I can tell, setup_build_envionment is never called, and I wasn't sure why.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you upload spack-build-env-mods.txt? It should be in <prefix>/.spack

**Comment by robertu94** [comment]
[spack-build-env-mods.txt](https://github.com/spack/spack/files/11289712/spack-build-env-mods.txt)

**Comment by adamjstewart** [comment]
That's odd, everything looks right, no idea why those env vars wouldn't be set. Maybe @alalazo knows something.

File: var/spack/repos/builtin/packages/py-cupy/package.py [hunk 1]
```diff
@@ -32,3 +32,12 @@ class PyCupy(PythonPackage):
     depends_on("nccl")
     depends_on("cudnn")
     depends_on("cutensor")
+
+    conflicts("~cuda")
+
+    def setup_build_envionment(self, env):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-cupy/package.py]:
```suggestion
    def setup_build_environment(self, env):
```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cupy/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/py-cupy/package.py]: :point_up: should hopefully fix your issue

Comment (robertu94) [reply var/spack/repos/builtin/packages/py-cupy/package.py]: Thanks good catch.  After fixing this, I found one more bug.  should be ready to merge now.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@adamjstewart please go ahead merging if this looks good to you

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**